miner/worker.go

```diff
@@ -585,48 +582,37 @@ func (w *worker) resultLoop() {
 			}
 			// Different block could share same sealhash, deep copy here to prevent write-write conflict.
 			var logs []*types.Log
-			work := w.current
 
-			for _, receipt := range append(work.receipts, work.privateReceipts...) {
+			for _, receipt := range append(task.receipts, task.privateReceipts...) {
 				// Update the block hash in all logs since it is now available and not when the
 				// receipt/log of individual transactions were created.
 				for _, log := range receipt.Logs {
 					log.BlockHash = hash
 				}
-			}
-
-			for _, log := range append(work.state.Logs(), work.privateState.Logs()...) {
-				log.BlockHash = hash
+				logs = append(logs, receipt.Logs...)
 			}
 
 			// write private transacions
-			privateStateRoot, _ := work.privateState.Commit(w.config.IsEIP158(block.Number()))
+			privateStateRoot, _ := task.privateState.Commit(w.config.IsEIP158(block.Number()))
 			core.WritePrivateStateRoot(w.eth.ChainDb(), block.Root(), privateStateRoot)
-			allReceipts := mergeReceipts(work.receipts, work.privateReceipts)
+			allReceipts := mergeReceipts(task.receipts, task.privateReceipts)
 
 			// Commit block and state to database.
-			w.mu.Lock()
-			stat, err := w.chain.WriteBlockWithState(block, allReceipts, work.state, nil)
-			w.mu.Unlock()
+			stat, err := w.chain.WriteBlockWithState(block, allReceipts, task.state, nil)
 			if err != nil {
-				log.Error("Failed writWriteBlockAndStating block to chain", "err", err)
```

Comment:
we should log error if  `w.chain.WriteBlockWithState(...)` fails. upstream does the same